features/polyExplorer/src/components/entityList/entityList.jsx

```diff
@@ -5,46 +5,49 @@ import EntityShortInfo from "../entityShortInfo/entityShortInfo.jsx";
 
 import "./entityList.css";
 
-function getStartGroups(entities) {
-    let numberGroups = 0;
-    let numberValues = 0;
-    const keys = Object.keys(entities);
-    for (let e of keys) {
-        numberGroups++;
-        numberValues += entities[e].length;
-        if (numberValues > 15) return keys[numberGroups];
+function determineInitialGroups(entities) {
+    const groups = Object.keys(entities);
+    let remainingValues = 15;
+    for (let i = 0; i < groups.length; i++) {
+        remainingValues -= entities[groups[i]].length;
+        if (remainingValues < 0) return Math.min(i + 2, groups.length);
```

Comment:
It does eliminate a few magic numbers, or at least gives them a name. I don't know what to do with the "2", though...

Reply:
It returns the number of groups, while `i` is the index of the current group. So `i + 1` means "include the current group", while `i + 2` means "include the current group and the next". Given how this is index/size based logic, I don't think it'd become clearer if there was a named constant for this. I could do `i + 1 + extraGroupsToInclude` where the latter is defined as `2`, but dunno, I find that less clear actually :D The code is just a few lines, doesn't seem that hard to read.